proposal.md

```diff
@@ -0,0 +1,186 @@
+
+## Introduction
+
+There are two general areas where Node loads JavaScript and therefore needs a way to know whether to parse the JavaScript source code as CommonJS or as ESM:
+
+1. Initial entry points, such as running `node file.js` on the command line, or running `node` with  `--eval`, piping source code to Node via `STDIN`, or running an extensionless executable JavaScript file with a hashbang.
+
+2. `import` statements and `import()` expressions, where a file or package is specified as the source to be imported, e.g. `import _ from 'lodash'`.
+
+This proposal aims to cover how Node should decide what module system to use for each of the use cases within the above groups. This is the _interpretation_ part of module resolution; the other part is _location._ Locating the code to be interpreted is covered by the [resolver specification][nodejs/ecmascript-modules:esm.md#resolver-algorithm], and involves things like taking the specifier string `'underscore'` and finding its package entry point file `./node_modules/underscore/underscore.js`. Location is out of scope for this proposal. This proposal covers what Node should do once the source code to be parsed has already been located.
+
+A central new concept used by many of the methods suggested in the proposal is the _package scope,_ delineated by _package boundaries._
+
+## Package scopes and package boundaries
+
+A _package scope_ is a folder containing a `package.json` file and all of that folder’s subfolders except those containing another `package.json` file and _that_ folder’s subfolders. A folder containing a `package.json` is a _package boundary._ For example:
+
+```
+/usr/                        - outside any package scope
+/usr/src/                    - outside any package scope
+
+/usr/src/app/package.json    - in “app” package scope
+/usr/src/app/index.js        - in “app” package scope
+/usr/src/app/startup/init.js - in “app” package scope
+/usr/src/app/node_modules/   - in “app” package scope
+
+/usr/src/app/node_modules/sinon/package.json - in “sinon” package scope
+/usr/src/app/node_modules/sinon/lib/sinon.js - in “sinon” package scope
+```
+
+The `package.json` files here each create a new package scope for the folder they’re in, and all subfolders down until another `package.json` file creates a new scope below. Thus `/usr/src/app/` and `/usr/src/app/node_modules/sinon/` are each package boundaries, with different package scopes above and below.
+
+Within a package scope, `package.json` files can hold metadata and configuration for the files contained within that scope. This proposal uses this concept to allow `package.json` values to opt in a package into being interpreted as ESM.
+
+Package scopes apply only to files. While a user may execute code via `node --eval` from within a working directory of a particular package scope, that package scope has no effect on the source code passed in via `node --eval`.
+
+## Package scope algorithm
```

Comment:
```suggestion
## Algorithm for Determination of Mode based on Package Scope 
```

Mode can be whatever we [bikeshed above](https://github.com/GeoffreyBooth/node-import-file-specifier-resolution-proposal/pull/12#pullrequestreview-199516278)